src/main/java/dinkplugin/util/Utils.java

```diff
@@ -87,6 +89,18 @@ public String getChatBadge(@NotNull AccountType type) {
         }
     }
 
+    public BufferedImage rescale(BufferedImage input, double percent) {
+        if (percent + Math.ulp(1.0) >= 1.0)
+            return input;
+
+        AffineTransform rescale = AffineTransform.getScaleInstance(percent, percent);
+        AffineTransformOp operation = new AffineTransformOp(rescale, AffineTransformOp.TYPE_BILINEAR);
```

Comment:
Other options include nearest-neighbor (low quality) and bicubic (slow). Bilinear strikes a good performance-quality balance, and this approach is faster than [Image#getScaledInstance](https://docs.oracle.com/javase/8/docs/api/java/awt/Image.html#getScaledInstance(int,%20int,%20int)) (which has other [problems](https://web.archive.org/web/20200805115809/https://community.oracle.com/docs/DOC-983611), but is still used by [RuneLite](https://github.com/runelite/runelite/blob/ed8076f4d18efe90f59aeedc4c1ef412d9f65b7b/runelite-client/src/main/java/net/runelite/client/util/ImageUtil.java#L259))

Reply:
This choice was decided by the fact that the first attempt literally made the sizes larger 🙃